pkg/util/kubernetes/hostinfo/tags.go

```diff
@@ -90,7 +93,10 @@ func getLabelsToTags() map[string]string {
 
 func getAnnotationsToTags() map[string]string {
 	annotationsToTags := map[string]string{}
-	for k, v := range config.Datadog().GetStringMapString("kubernetes_node_annotations_as_tags") {
+
+	metadataAsTags := configutils.GetMetadataAsTags(config.Datadog())
```

Comment:
Are these two functions called at init only or pulled periodically in workloadmeta?

Reply:
Yes you are right, good catch, these functions are not called only at startup, so might be better to avoid computing the config each time. 

I added this commit to ensure it is computed only once, but it includes introducing a global state. 

If we don't want to present a global state, we will need to do some non-negligible refactoring to the tags package. 

WDYT?